### PR TITLE
feat:  implementation of an AI Assistant feature within the request  API.

### PR DESCRIPTION
**Pull request by itsmdasifraza**
## Description
This pull request introduces the implementation of an AI Assistant feature within the API tool. The AI Assistant leverages machine learning models to provide users with intelligent suggestions, automated responses, and enhanced data analysis capabilities.
#1240 #1241

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @itsmdasifraza on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 05:22:05 (UTC+0)_

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @itsmdasifraza on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 06:50:18 (UTC+0)_

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @Astitva877 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 09:34:23 (UTC+0)_

File: src/packages/@workspaces/features/chat-bot/components/ai-chat-interface/AIChatInterface.svelte [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+   */
+  $: scrollList = (_param: "bottom", _pixels, _behaviour) =>
+    scroll(_param, _pixels, _behaviour);
+</script>
+
+<div class="d-flex flex-column h-100 chat-box">
+  <!-- <div
+  class="d-flex flex-column h-100 chat-box"
+  transition:slide={{ duration: 400 }}
+> -->
```
Comment (Astitva877) [inline src/packages/@workspaces/features/chat-bot/components/ai-chat-interface/AIChatInterface.svelte:57]: remove this if not needed

File: src/packages/@workspaces/features/chat-bot/components/ai-suggestion-box/AISuggestionBox.svelte [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    height: 28px;
+    font-size: 16px;
+    cursor: pointer;
+    transition:
+      background-color 0.3s,
+      box-shadow 0.3s;
+    margin-bottom: 8px;
+    right: 0;
+    /* width: fit-content; */
+    /* text-align: right; */
```
Comment (Astitva877) [inline src/packages/@workspaces/features/chat-bot/components/ai-suggestion-box/AISuggestionBox.svelte:42]: not needed

File: src/packages/@workspaces/features/chat-bot/components/chat-item/ChatItem.svelte [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+      .message-wrapper .markdown li,
+      .message-wrapper .markdown a,
+      .message-wrapper .markdown span,
+      .message-wrapper .markdown code
+    ) {
+    font-size: 12px;
+  }
+  :global(.message-wrapper .markdown pre) {
+    /* background-color: black; */
+    /* padding: 10px; */
```
Comment (Astitva877) [inline src/packages/@workspaces/features/chat-bot/components/chat-item/ChatItem.svelte:239]: not needed

File: src/packages/@workspaces/features/chat-bot/layout/ChatBot.svelte [hunk 4]
```diff
@@ -0,0 +1,173 @@
+<script lang="ts">
+  import type { Observable } from "rxjs";
+  // import { fade } from "svelte/transition";
```
Comment (Astitva877) [inline src/packages/@workspaces/features/chat-bot/layout/ChatBot.svelte:3]: remove this

File: src/packages/@workspaces/features/chat-bot/layout/ChatBot.svelte [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        width={"18px"}
+        color="var(--icon-primary-300)"
+      />
+    </div>
+    <div class="d-flex flex-column align-items-end">
+      <!-- <div
+      class="d-flex flex-column align-items-end"
+      in:fade={{ duration: 200 }}
+      out:fade={{ duration: 200 }}
+    > -->
```
Comment (Astitva877) [inline src/packages/@workspaces/features/chat-bot/layout/ChatBot.svelte:112]: remove this  if not needed

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @Astitva877 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 11:15:33 (UTC+0)_

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @itsmdasifraza on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 12:02:53 (UTC+0)_

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @Astitva877 on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 12:15:54 (UTC+0)_

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @itsmdasifraza on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 12:39:53 (UTC+0)_

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @itsmdasifraza on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/30/2024, 12:53:25 (UTC+0)_

**Review by Astitva877 — APPROVED** [review]

**Comment by pratikshinde590** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @itsmdasifraza on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|New reliability rating|:white_check_mark: OK|1|> 1|
|New security rating|:white_check_mark: OK|1|> 1|
|New maintainability rating|:white_check_mark: OK|1|> 1|

[View on SonarQube](https://sonar.techdomeaks.com/dashboard?id=sparrow)
###### _updated: 7/31/2024, 06:17:14 (UTC+0)_

**Pull request merged**